math/gts/Portfile

```diff
@@ -19,17 +18,27 @@ long_description    GTS stands for the GNU Triangulated Surface Library. \
                     3D surfaces meshed with interconnected triangles.
 
 homepage            http://gts.sourceforge.net/
-master_sites        sourceforge:project/${name}/${name}/${version}
+distname            ${name}-snapshot-${version}
+master_sites        http://gts.sourceforge.net/tarballs
 
-checksums           sha1    000720bebecf0b153eb28260bd30fbd979dcc040 \
-                    rmd160  2bfd4121412c16d63ee0aca65ccbc2601d6a7d51
+checksums           rmd160  24b3c469d5a35b658825eb342ae03e9e9d93afb8 \
+                    sha256  c23f72ab74bbf65599f8c0b599d6336fabe1ec2a09c19b70544eeefdc069b73b \
+                    size    880569
 
 depends_build       port:pkgconfig
 
-depends_lib         path:lib/pkgconfig/glib-2.0.pc:glib2 \
-                    port:libnetpbm
+depends_lib         path:lib/pkgconfig/glib-2.0.pc:glib2
 
-patchfiles          patch-examples-happrox.c.diff
+depends_test        port:libnetpbm
+
+test.run            yes
+test.target         check
+post-extract {
+    # test files do not have the correct permissions
+    foreach file [glob ${worksrcpath}/test/**/*.sh] {
+        system "chmod +x ${file}"
```

Comment:
It would be better to use `fs-traverse` rather than `foreach file` `glob`, and to use `file attributes` `-permissions` rather than `system` `chmod`.

Reply:
I added `file attributes -permissions`.
Thank you for the improvement.

Could you say a little more about why `fs-traverse` is superior?
I see how it could be used, but then we would have to test each file to see if it matches the pattern `*.sh`.
Isn't that exactly what `glob` is for?
Please forgive me if I am missing something.

Reply:
Since I haven't heard any objections, I will go ahead and merge this.
If need be, the `glob`->`fs-traverse` issue can be fixes without a revbump.

Reply:
`glob` is fine. I was mis-seeing this code as running an external executable, which is what we want to avoid when we can implement it directly in Tcl.